core/src/memory.rs

```diff
@@ -68,68 +63,83 @@ static_assertions::const_assert_eq!(WASM_PAGE_SIZE % GEAR_PAGE_SIZE, 0);
 static_assertions::const_assert_eq!(WASM_PAGE_SIZE % PAGE_STORAGE_GRANULARITY, 0);
 static_assertions::const_assert_eq!(PAGE_STORAGE_GRANULARITY % GEAR_PAGE_SIZE, 0);
 
+/// Alias for inner type of page buffer.
+pub type PageBufInner = LimitedVec<u8, (), GEAR_PAGE_SIZE>;
```

Comment:
This is not necessary from my point of view. Since PageBuf is kept in Substrate's StorageMap after the PR it appears in metadata but polkadot.js doesn't support fixed size array longer 2048 items. That is the reason of such solution and it looks like there is no other ways.

Comments about the case are several lines below.

Reply:
I'm aware of performance decrease here